apps/wallet-mobile/src/features/Receive/useCases/EnterAmountScreen.tsx

```diff
@@ -0,0 +1,149 @@
+import {SkeletonAdressDetail} from '../common/SkeletonAddressDetail/SkeletonAddressDetail'
+import {useStrings} from '../common/useStrings'
+
+export const EnterAmountScreen = () => {
+  const strings = useStrings()
+  const wallet = useSelectedWallet()
+  const receiveAddresses = useReceiveAddresses(wallet)
+
+  const HEIGHT_SCREEN = useWindowDimensions().height
+  const HEIGHT_MODAL = (HEIGHT_SCREEN / 100) * 80
+
+  const [amount, setAmount] = React.useState<string>('')
+
+  const {openModal} = useModal()
+
+  const {colors, styles} = useStyles()
+
+  const currentAddress = _.last(receiveAddresses)
+
+  const generateLink = React.useCallback(() => {
+    Keyboard.dismiss()
+
+    openModal(strings.multipleAdress, <Modal amount={amount} />, HEIGHT_MODAL)
+  }, [HEIGHT_MODAL, amount, openModal, strings.multipleAdress])
+
+  return (
+    <SafeAreaView style={styles.root} edges={['left', 'right', 'bottom']}>
+      <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
+        <KeyboardAvoidingView style={styles.root}>
+          <Spacer height={24} />
+
+          <View style={styles.content}>
+            <View style={styles.screen}>
+              <Text style={styles.textAddressDetails}>{strings.specificAmountDescription}</Text>
+
+              <TextInput label={strings.ADALabel} keyboardType="number-pad" onChangeText={setAmount} maxLength={4} />
+
+              <View style={styles.textSection}>
+                <Text style={[styles.textAddressDetails, {color: colors.gray}]}>{strings.address}</Text>
+
+                <Text style={styles.textAddressDetails}>{currentAddress}</Text>
+              </View>
+            </View>
+
+            <Button
+              shelleyTheme
+              onPress={generateLink}
+              disabled={amount === '' ? true : false}
+              title={strings.generateLink}
+              style={styles.button}
+            />
+
+            <Spacer height={24} />
+          </View>
+        </KeyboardAvoidingView>
+      </TouchableWithoutFeedback>
+    </SafeAreaView>
+  )
+}
+
+const Modal = ({amount}: {amount: string}) => {
+  const strings = useStrings()
+  const {styles} = useStyles()
+  const [isCopying, copy] = useCopy()
+
+  return (
+    <View style={styles.root}>
+      <ScrollView>
+        {mocks.specificAddressAmount !== null ? (
+          <ShareQRCodeCard
+            title={`${amount} ADA`}
+            address={mocks.specificAddressAmount}
+            onLongPress={() => copy(mocks.specificAddressAmount)}
+          />
+        ) : (
+          <View style={styles.root}>
+            <SkeletonAdressDetail />
+          </View>
+        )}
+
+        <Spacer height={32} />
+      </ScrollView>
+
+      <Button
+        shelleyTheme
+        onPress={() => {
+          copy(mocks.specificAddressAmount)
+        }}
+        disabled={amount === '' ? true : false}
+        title={strings.copyLinkBtn}
+        iconImage={require('../../../assets/img/copy.png')}
```

Comment:
can anything be done here, or is it legacy, -> SVG?

Reply:
I see that is legacy across the app as we have this inside the Button component 
```
        {iconImage != null && <Image source={iconImage} />}
```
and more components are like this 😖